src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py

```diff
@@ -199,20 +209,88 @@ def process(
                 if grant_statements is not None:
                     cc.message(grant_statements)
                     collected_output.append(grant_statements)
+            extension_fns_ddl = "\n".join(collected_output)
+            dest_file_py_file_to_ddl_map[py_file] = extension_fns_ddl
+
+        write_to_sql = kwargs.get(
+            "write_to_sql", True
+        )  # temporary until we have better e2e testing mechanism
+
+        if write_to_sql:
+            dest_file_to_sql_file = create_and_write_to_sql_files(
+                dest_file_py_file_to_ddl_map=dest_file_py_file_to_ddl_map,
+                deploy_root=self.deploy_root,
+                generated_root=self.generated_root,
```

Comment:
nit: combine into just a destination directory? No need to have other parts of the logic worry about what the relationship of generated root is to the deploy root. This way it would be really easy to introduce subdirs of the generated root in the future, for example.